R/add_cols.R

```diff
@@ -0,0 +1,247 @@
+        age_bracket <- risk$min_age[i]:risk$max_age[i]
+        if (hosp) {
+          age_group <- which(
+            .data$age %in% age_bracket & !is.na(.data$hospitalisation)
+          )
+        } else {
+          age_group <- which(
+            .data$age %in% age_bracket & is.na(.data$hospitalisation)
+          )
+        }
+        not_hosp_death_prob <- 1 - risk$risk[i]
+        age_group_sample <- sample(
+          age_group,
+          replace = FALSE,
+          size = not_hosp_death_prob * length(age_group)
+        )
+        .data$deaths[age_group_sample] <- NA
+      }
+    }
+    .data
+  }
+
+  .data <- apply_death_risk(.data, risk = hosp_death_risk, hosp = TRUE)
+  .data <- apply_death_risk(.data, risk = non_hosp_death_risk, hosp = FALSE)
+
+  # return data
+  .data
+}
+
+
+#' Add line list information as column to infectious history `<data.frame>`
+#'
+#' @param .data A `<data.frame>` containing the infectious history from a
+#' branching process simulation
+#' @param distribution A `character` with the name of the distribution,
+#' following the base R convention for distribution naming (e.g. Poisson
+#' is `pois`).
+#' @param ... [dots] Extra arguments to be passed to the distribution function
+#' given in the `distribution` argument.
+#'
+#' @name .add_info
+#'
+#' @return A `<data.frame>` with one more column than input into `.data`.
+#' Unless the column heading is already present in which the data is
+#' overwritten.
+#' @keywords internal
+#'
+NULL
+
+#' @name .add_info
+.add_names <- function(.data) {
+  .data$case_name <- .sample_names(.data = .data)
+
+  # left join corresponding names to infectors preserving column and row order
+  infector_names <- data.frame(id = .data$id, infector_name = .data$case_name)
+  col_order <- c(colnames(.data), "infector_name")
+  .data <- merge(
+    .data, infector_names,
+    by.x = "infector", by.y = "id", all.x = TRUE
+  )
+  .data <- .data[order(is.na(.data$infector_name), decreasing = TRUE), ]
+  .data <- .data[col_order]
+  row.names(.data) <- NULL
+
+  # return named line list
+  .data
+}
+
+#' @name .add_info
+.add_ct <- function(.data, distribution = c("norm", "lnorm"), ...) {
+  distribution <- match.arg(distribution)
+
+  rdist <- switch(distribution,
+    norm = stats::rnorm,
+    lnorm = stats::rlnorm
+  )
+
+  # c() over ...length() to ensure NULL is not counted by length
+  if (length(c(...)) == 0) {
+    stop("Distribution parameters are missing, check config", call. = FALSE)
+  }
+
+  # name list elements with vec names to ensure arg matching in do.call
+  args <- c(n = 1, list(...))
+
+  ct_value <- tryCatch(
+    error = function(cnd) {
+      stop(
+        "Incorrect parameterisation of distribution, check config",
+        call. = FALSE
+      )
+    },
+    warning = function(cnd) {
+      stop(
+        "Incorrect parameterisation of distribution, check config",
+        call. = FALSE
+      )
+    },
+    do.call(rdist, args = args)
+  )
```

Comment:
Prefer pattern of try then catch, although not really an issue

Reply:
Resolved already in #77.